libs/ts-rest/core/src/lib/type-utils.ts

```diff
@@ -40,11 +40,11 @@ export type GetFieldType<T, P> = P extends `${infer Left}.${infer Right}`
 // https://stackoverflow.com/questions/63447660/typescript-remove-all-properties-with-particular-type
 // Nested solution also available ^
 type ExcludeKeysWithTypeOf<T, V> = {
-  [K in keyof T]: Exclude<T[K], undefined> extends V ? never : K;
+  [K in keyof T]-?: Exclude<T[K], undefined> extends V ? never : K;
```

Comment:
@oliverbutler `Without<DataReturnArgs, never>` was not working and returning `unknown` because the new `headers` argument is optional and these mapped types did not include optional keys. This is what was triggering everything to become optional.